lib/components/rest.ts

```diff
@@ -305,29 +305,38 @@ export default function createRestAction<Context extends GatewayContext>(
             const expectedResponseContentType =
                 config.expectedResponseContentType || options.expectedResponseContentType;
 
-            if (
-                actualResponseContentType &&
-                expectedResponseContentType &&
-                actualResponseContentType !== expectedResponseContentType
-            ) {
-                ctx.log('Invalid response content type', {
-                    expectedResponseContentType,
-                    actualResponseContentType,
-                });
-                ctx.end();
+            if (actualResponseContentType && expectedResponseContentType) {
+                let isInvalidResponseContentType;
```

Comment:
It might have been look a little simpler if you've checked for valid response content type instead of invalid (since not- checks are harder to read than positive ones).

i.e.

```typescript
isValidResponseContentType = false

isValidResponseContentType = expectedResponseContentType.includes(String(actualResponseContentType));

isValidResponseContentType = expectedResponseContentType === actualResponseContentType
```